code/Model/SiteTree.php

```diff
@@ -2100,8 +2100,12 @@ public function getSettingsFields()
             }
         }
 
+        $inheritMessage = $this->ParentID !== 0 ?
```

Comment:
I think strict type checking on DB fields is a bad idea as I'm pretty sure they are all loaded as strings